test/gtest/argmax.cpp

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "argmax.hpp"
+#include <miopen/env.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+std::string GetFloatArg()
+{
+    const auto& tmp = miopen::GetStringEnv(ENV(MIOPEN_TEST_FLOAT_ARG));
+    if(tmp.empty())
+    {
+        return "";
+    }
+    return tmp;
+}
+
+struct ArgmaxTestFloat : ArgmaxTest<float>
+{
+};
+
+TEST_P(ArgmaxTestFloat, ArgmaxTestFw)
+{
+    if(miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)) && (GetFloatArg() == "--float"))
```

Comment:
The general requirement is that if a gtest is run without env vars (i.e. out of the context of _ctest_), then it should test all supported datatypes (FLOAT, HALF etc), and _as if_ `MIOPEN_TEST_ALL` is set to On.
```suggestion
    if(!miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)) && (GetFloatArg() == "--float" || GetFloatArg().empty()))
```

Reply:
fixed it.

Reply:
[resolved]

Reply:
This change was expected to cause GitHub actions to fail, so it was reverted. After that, the github action passed, but it is difficult for me to determine the detailed cause. This is because I cannot confirm the results of CI/CD.

Reply:
@CAHEK7 This is [unresolved]

Reply:
@seungmanhan 
> This change was expected to cause GitHub actions to fail, so it was reverted. 

Excuse me, but I do not fully understand. "Expected to cause failures" or "caused failures"? I guess the latter but please confirm.

Something is wrong. The suggested change should NOT lead to any failures on CI. Before merging, we must triage the CI failure and find the root reason (it might just be a random failure).

@CAHEK7 Can you please find a log of the failing CI stage of commit 4873cc0, zip and attach it to the reply?